reyna/src/com/b2msolutions/reyna/Repository.java

```diff
@@ -31,36 +33,92 @@ public void onCreate(SQLiteDatabase db) {
 
 	@Override
 	public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
-        Logger.v(TAG, "onUpgrade");
+		Logger.v(TAG, "onUpgrade");
 	}
 
 	public void insert(Message message) {
-        Logger.v(TAG, "insert");
-		if (message == null)
+		Logger.v(TAG, "insert");
+		if (message == null) {
+			return;
+		}
+
+		SQLiteDatabase db = this.getWritableDatabase();
+		this.insertMessage(db, message);
+	}
+
+	public void insert(Message message, long dbSizeLimit) {
+		Logger.v(TAG, "insert");
+		if (message == null) {
 			return;
+		}
+
+		SQLiteDatabase db = this.getWritableDatabase();
+		long dbSize = this.getDbSize(db);
+
+		//todo handle (dbSize > dbSizeLimit) with a shrink db operation instead
+		//of doing just one deletion
+		if (dbSize > dbSizeLimit || Math.abs(dbSizeLimit - dbSize) < (dbSize * 0.2)) { //0.2 = 20% of the db size
```

Comment:
We might need only the threshold check here.
Can we have this in a small separate function called for example shouldReplace()

@Youhana-Hana